Update Theme Docs (#656)
reflex-dev/reflex-web: docs/styling/theming.md

To access a specific shade of color from the theme, you can use the `rx.color`. When switching to light and dark themes, the color will automatically change. Shades can be accessed by using the color name and the shade number. The shade number ranges from 1 to 12. Additionally, they can have their alpha value set by using the `True` parameter it defaults to `False`. A full list of colors can be found [here](https://www.radix-ui.com/colors).

```python demo
rx.flex(
    rx.button(
        "Hello World",
        color=rx.color("grass", 1),
        background_color=rx.color("grass", 7),
        border_color=f"1px solid {rx.color('grass', 1)}",
    ),
    spacing="2"
)
```

```python eval
rx.table.root(
    rx.table.header(
        rx.table.row(
            rx.table.column_header_cell(rx.code("Name")),
            rx.table.column_header_cell("Type"),
            rx.table.column_header_cell("Description"),
        ),
    ),
    rx.table.body(
        rx.table.row(
            rx.table.row_header_cell(rx.code("color")),
            rx.table.cell("Str"),

>             rx.table.cell("The color to use. Can be any color from the theme."),

Reviewer: ```suggestion
            rx.table.cell("The color to use. Can be any valid accent color or 'accent' to reference the current theme color."),
```

        ),
        rx.table.row(
            rx.table.row_header_cell(rx.code("shade")),
            rx.table.cell('1 - 12'),

>             rx.table.cell("The shade of the color to use. Defaults to 7."),

Reviewer: We should probably link to this https://www.radix-ui.com/colors/docs/palette-composition/understanding-the-scale

Reviewer: i was thinking that specifically linking to the "Understanding the Scale" page would help people decode the why behind the numbers

        ),
        rx.table.row(
            rx.table.row_header_cell(rx.code("alpha")),
            rx.table.cell("Bool"),

>             rx.table.cell("Whether tp use the alpha value of the color. Defaults to False."),

Reviewer: ```suggestion
            rx.table.cell("Whether to use the alpha value of the color. Defaults to False."),
```

    rx.button(
        "Hello World",
        color="white",
        background_color="#87CEFA",
        border="1px solid rgb(176,196,222)",
    ),
    spacing="2"
)
```

## Toggle Appearance

To toggle between the light and dark mode manually, you can use the `toggle_color_mode` with the desired event trigger of your choice. 

```python 

from reflex.style import toggle_color_mode

def index():
    return rx.button(
        "Toggle Color Mode",
        on_click=toggle_color_mode,
    )
```

## Appearance Conditional Rendering

To render a different component depending on whether the app is in `light` mode or `dark` mode, you can use the `rx.color_mode_cond` component. The first component will be rendered if the app is in `light` mode and the second component will be rendered if the app is in `dark` mode.

```python demo

> rx.color_mode_cond(
>     rx.image(src="/logos/light/reflex.svg", height="4em"),
>     rx.image(src="/logos/dark/reflex.svg", height="4em"),
> )

Reviewer: i think we should show this with the kwargs

```suggestion
rx.color_mode_cond(
    light=rx.image(src="/logos/light/reflex.svg", height="4em"),
    dark=rx.image(src="/logos/dark/reflex.svg", height="4em"),
)
```
